### PR TITLE
Add doc how to update Istio manifest

### PR DESCRIPTION
**Pull request by nak3**
This patch adds the doc how to update Istio manifest under third_party.

### CODE REVIEW

File: ISTIO.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  The script updates manifest automatically so you just kick it:
+
+  ```sh
+  ./hack/update-codegen.sh
+  ```
+
+#### 3. Send the PR
+
+  Once the manifest was updated, please send the PR against net-istio repository. The CI starts using the updated manifest and detect any issue if exists.
```
Comment (AngeloDanducci) [inline ISTIO.md:41]: Recommend changing `manifest was updated` to `manifest is updated`.

Also recommend changing `detect` to `detects`

File: ISTIO.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  The istio-head manifest script is commented out in `hack/update-codegen.sh` to avoid frequent update.
+  You can edit `hack/update-codegen.sh` and uncommnet the script as:
+
+  ```sh
+  ${REPO_ROOT_DIR}/third_party/istio-head/generate-manifests.sh
+  ```
+
+#### 2. Run `hack/update-codegen.sh`
+
+  The script updates manifest automatically so you just kick it:
```
Comment (AngeloDanducci) [inline ISTIO.md:58]: Recommend changing `updates manifest` to `updates the manifest`

**Review by AngeloDanducci — COMMENTED** [review]
Great update, makes it very clear and easy how to bump versions 👍

**Comment by nak3** [comment]
Thank you! Updated.

> One other thing to consider adding is when/why we would want to update the head version.

That's good point. We definitely should decide/improve the process. I opened https://github.com/knative-sandbox/net-istio/issues/913 and add the `TODO` in the doc.

**Comment by AngeloDanducci** [comment]
Seems odd that this would fail given none of these changes affect the underlying code.

Are those tests known to be flaky?

**Comment by dprotaso** [comment]
/retest-required

yeah mesh tests are flakey - https://testgrid.k8s.io/r/knative-own-testgrid/serving


@nak3 probably has more insight into why

**Pull request merged**